File: c/difference-of-squares/difference_of_squares.c
#include "difference_of_squares.h"

unsigned int sum_of_squares(unsigned int number)
{
    int ret;
    for (ret = 0; number > 0;number--) {
        ret += number * number;
    }
    return ret;
}

unsigned int square_of_sum(unsigned int number)
{
    int sum;
    for (sum = 0; number > 0;number--) {
        sum += number;
    }
    return sum * sum;
}

unsigned int difference_of_squares(unsigned int number)
{
    return square_of_sum(number) - sum_of_squares(number);
}

File: dart/difference-of-squares/lib/difference_of_squares.dart
class DifferenceOfSquares {
  int squareOfSum(int number) {
    int result = 0;
    for (int i = 1; i <= number; i++) {
      result += i;
    }
    return result * result;
  }

  int sumOfSquares(int number) {
    int result = 0;
    for (int i = 1; i <= number; i++) {
      result += i * i;
    }
    return result;
  }

  int differenceOfSquares(int number) {
    return squareOfSum(number) - sumOfSquares(number);
  }
}
